Data/HashMap/Internal.hs

```diff
@@ -196,8 +205,10 @@ import Data.Coerce (coerce)
 ------------------------------------------------------------------------
 
 -- | Convenience function.  Compute a hash value for the given value.
-hash :: H.Hashable a => a -> Hash
-hash = fromIntegral . H.hash
+hash ::forall a (salt :: Nat) . H.Hashable a => KnownNat salt => Proxy salt -> a -> Hash
```

Comment:
Is there a reason for the double `=>` rather than the conventional constraint tuple? If so, please add an explanatory comment. Also, is there a reason you use `Proxy` here and not `Proxy#`? I'd prefer the latter to be more confident we won't pass an actual run-time value.